c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -1290,7 +1337,28 @@ class PostgresCopyStreamWriter {
       }
       const ArrowType arrow_type = schema_view.type;
       PostgresCopyFieldWriter* child_writer;
-      NANOARROW_RETURN_NOT_OK(MakeCopyFieldWriter(arrow_type, &child_writer, error));
+      if (arrow_type == NANOARROW_TYPE_TIMESTAMP) {
+        switch (schema_view.time_unit) {
+          case NANOARROW_TIME_UNIT_NANO:
+            child_writer = new PostgresCopyTimestampFieldWriter<
+              NANOARROW_TIME_UNIT_NANO>();
+            break;
+          case NANOARROW_TIME_UNIT_MILLI:
+            child_writer = new PostgresCopyTimestampFieldWriter<
+              NANOARROW_TIME_UNIT_MILLI>();
+            break;
+          case NANOARROW_TIME_UNIT_MICRO:
+            child_writer = new PostgresCopyTimestampFieldWriter<
+              NANOARROW_TIME_UNIT_MICRO>();
+            break;
+          case NANOARROW_TIME_UNIT_SECOND:
+            child_writer = new PostgresCopyTimestampFieldWriter<
+              NANOARROW_TIME_UNIT_SECOND>();
+            break;
+        }
+      } else {
+        NANOARROW_RETURN_NOT_OK(MakeCopyFieldWriter(arrow_type, &child_writer, error));
```

Comment:
I decided it was better to not try and add the `ArrowTimeUnit` argument to the factory function and handle separately, but open to either approach